### PR TITLE
Adds initial CONTRIBUTING guidelines

### PR DESCRIPTION
**Pull request by ellistarn**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/37)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Contributor Guide
+
+## Setting up a KFServing Environment
+TODO flesh this out
+1. make all
+2. make docker-build
+3. make docker-push
+4. make deploy
+5. make install
+4. kubectl apply -f docs/samples/tensorflow.yaml
```
Comment (animeshsingh) [inline CONTRIBUTING.md]: Believe this should be part of developer guide, and we should follow the general convention for CONTRIBUTING guidelines for rest of the repos here.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+1. dep ensure
+2. git commit -am "Updates vendor dependencies"
+
+## Submitting a PR
+The following should be viewed as Best Practices unless you know better ones (please submit a guidelines PR).
+
+| Practice | Rationale |
+| ----- | --------- |
+| Keep the code clean | The health of the codebase is imperative to the success of the project. Files should be under 500 lines long in most cases, which may mean a refactor is necessary before adding changes. |
+| Squash your commits | A clean codebase needs a clean commit log. Aim to submit one commit per PR, which may mean squashing your changes. You'll thank yourself later if you ever need to revert changes you've made and you'll simplify the lives of anyone looking into the history of the project. |
```
Comment (abhi-g) [inline CONTRIBUTING.md]: Does PR merge process automatically squash commits in a PR when creating a merge commit?

Comment (animeshsingh) [reply CONTRIBUTING.md]: I belive it does @abhi-g

Comment (ellis-bigelow) [reply CONTRIBUTING.md]: oh awesome. I guess your commit messages just get squashed instead of fixup

File: CONTRIBUTING.md (hunk 1 above)
Comment (ellis-bigelow) [reply CONTRIBUTING.md]: SGTM

**Pull request merged**